File: algorithm-toolbox/greatest-common-devisor/file.py
def gcd(i,j):
    if i == 0:
        return j
    if j == 0:
        return i
    return gcd(i%j,j) if i>j else gcd(i,j%i)
print("Enter two space separated Integers")
i , j = [int(x) for x in input().split()]
print(gcd(i,j))

File: algorithm-toolbox/greatest-common-devisor/file.cpp
#include <iostream>
using namespace std;

int gcd(int i, int j)
{
    if (i == 0)
    {
        return j;
    }

    if (j == 0)
    {
        return i;
    }

    return i > j ? gcd(i % j, j) : gcd(i, j % i);
}

int main()
{
    int i, j;
    cout << "Enter two space separated integers" << endl;
    cin >> i >> j;
    cout << gcd(i, j) << endl;
    return 0;
}